### ISSUE TITLE
API: api/index/detail/world

### ISSUE THREAD

**Issue by nicHoch** [theCrag/website#2206]
- [ ] Remove any routes/annotations for the world index call by default
- [ ] add the RoutesGradeBand information (histogram over all child routes as gradeBand)

**Comment by scd** [comment]
> Remove any routes/annotations for the world index call by default

Done. The api will no longer get routes that are not in a top level crag. In otherwords routes associated with regions will not really be part of the system and should be moved. This seems like a reasonable business rule to me.

> add the RoutesGradeBand information (histogram over all child routes as gradeBand) 

I have implemented two new query tags BandProfile and AscentBandProfile

@brendanheywood please let nicky know when you have pulled from repo so he can test.

@nicHoch please close this issue when you are happy. BTW, any of us can close issues if we feel that sufficient has been done to close it. Others can, and sometimes do, re-open if they feel there is more to do.

**Comment by nicHoch** [comment]
If i use BandProfile in the world Index query in the withdata field the api crashes

**Comment by scd** [comment]
Which end point. I may have implemented it in a different end point.

**Comment by nicHoch** [comment]
https://dev.thecrag.com/api/index/detail/world?key=123&withdata=NodeID%2CParentID%2CLastUpdated%2CSiblingSequence%2CNodeType%2CName%2CPoint%2CAreaType%2CNumberAscents%2CNumberRoutes%2CNumberTopos%2CPopularity%2CTLC%2CAlternateName%2CDescription%2CTopo%2CBandProfile%2CTag&abbr=0

Am 25.05.2016 um 23:38 schrieb Simon Dale:
> Which end point. I may have implemented it in a different end point.

**Comment by scd** [comment]
@brendanheywood could you please pull again from repo and install on dev. It looks like this code did not make it to dev.

**Comment by nicHoch** [comment]
Hi we should bring a bit consistency into property names and notations.
For example all withdata parameters start upper case and go on with camel case. (NumberTopos).
If i use other api end points all properties start lover case and go on with camel case. 
There are also other names for the same property:

in withdata NodeType and AreaType is used. in general and other endpoints this is type and subType.

The problem is i would like the same template where ever possible (to show an area for example) independet if the data is loaded from a local worldIndex or online via a single "getNode" call.

so bottom line any property should have the same name and notation within all endpoints.

**Comment by scd** [comment]
camel case should be the standard for API as this is most friendly for perl hashes.

@nicHoch this requirement will get lost in this issue and probably deserves it's own issue. I don't have the time to do a general review of the whole API for quite some time, so maybe also create a task list of where they are appearing non standard.

**Comment by brendanheywood** [comment]
deployed
